plugin/pkg/admission/priority/admission.go

```diff
@@ -162,6 +163,15 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 				return fmt.Errorf("failed to get default priority class: %v", err)
 			}
 		} else {
+			pcName := pod.Spec.PriorityClassName
+			// Only allow system priorities in the system namespace. This is to prevent abuse or incorrect
+			// usage of these priorities. Pods created at these priorities could preempt system critical
+			// components.
+			for _, spc := range scheduling.SystemPriorityClasses() {
+				if spc.Name == pcName && pod.Namespace != metav1.NamespaceSystem {
```

Comment:
The user can lie here about the namespace. Please change it to `a.GetNamespace()`.

Reply:
I am not sure I understand, a is already cast to pod so pod.Namespace is same as a.GetNamespace().

Reply:
`a.GetNamespace()` is the admission attributes parsed from the namespace the user actually submitted the pod to

pod.Namespace is whatever namespace the user specified in their pod spec

Reply:
priority class is not mutable, correct? if so, this check should likely only be done on create.

Reply:
> priority class is not mutable, correct? if so, this check should likely only be done on create.

Yes and that is what is happening, isn't it?

Reply:
Avesh is right. Since `admitPod` is called only on pod creation request, this check applies to pod creation only.